src/plugins/irc-events/link.js

```diff
@@ -313,71 +310,89 @@ function getRequestHeaders(headers) {
 	return formattedHeaders;
 }
 
-function fetch(uri, headers, cb) {
-	let req;
+function fetch(uri, headers) {
+	// Stringify the object otherwise the objects won't compute to the same value
+	const cacheKey = JSON.stringify({uri, headers});
+	let promise = currentFetchPromises.get(cacheKey);
 
-	try {
-		req = request.get({
-			url: uri,
-			maxRedirects: 5,
-			timeout: 5000,
-			headers: getRequestHeaders(headers),
-		});
-	} catch (e) {
-		return cb(null, e);
+	if (promise) {
+		return promise;
 	}
 
-	const buffers = [];
-	let length = 0;
-	let limit = Helper.config.prefetchMaxImageSize * 1024;
+	const removeCache = () => currentFetchPromises.delete(cacheKey);
```

Comment:
I'm probably missing something, but any reason why you're creating this here and not just above where it's used, L391?

Reply:
I tried chained promises, forgot to move it back down.